### ISSUE TITLE
"500 Internal Privoxy Error" or "This site can’t be reached"

### ISSUE THREAD

**Issue by iclinux** [shadowsocksr/ShadowsocksX-NG#101]
**System and Shadowsocksx-NG-R8 version:**

- macOS 10.11.6 (15G1217)
- ShadowsocksX-NG-R8.dmg Version 1.3.9-R8 (7)

**Expected behavior:**

can visit google.com

**Actual behavior:**

cannot, and said:
=============
500 Internal Privoxy Error

Privoxy encountered an error while processing your request:

Could not load template file forwarding-failed or one of its included components.

Please contact your proxy administrator.
If you are the proxy administrator, please put the required file(s)in the (confdir)/templates directory. The location of the (confdir) directory is specified in the main Privoxy config file. (It's typically the Privoxy install directory, or /etc/privoxy/).
=============
or 
=============
This site can’t be reached

The webpage at https://www.facebook.com/ might be temporarily down or it may have moved permanently to a new web address.
=============

(N/A for feature requests)

**Steps to reproduce:**

reproduce when visit google.com or any other web site.

(N/A for feature requests)

**How often does this happen?**
each time
(N/A for feature requests)

**ss-local.log**

Please upload the ss-local.log
[ss-local.log.zip](https://github.com/shadowsocksr/ShadowsocksX-NG/files/770229/ss-local.log.zip)

 file here the file is in `~/Library/Logs`
1) Open 'Advanced Settings -> enable Verbose Mode'
2) Continue run `Shadowsocksx-NG-R8` for 5 minutes
3) Upload the `~/Library/Logs/ss-local.log` here (with or without compress)

**Application log**

Open the `Console.app` and search `Shadowsocksx-NG`
Copy paste the log here

**Crash Log**

If the app crashes and pop up a crash log, please copy and paste here

**Tips:**

- Always check for the latest update, some known bug has been fix in the latest release such as old version not support auth_aes_md5 auth_aes_sha1

- Beta version of macOS also may couse some problems

- ss-local compiled by macOS Sierra 10.12 not compatite with 10.11, some old release's ss-local are compiled using 10.12 will not work on 10.11, new release are try build in 10.10 to max fit the compatite

**Comment by iclinux** [comment]
BTW, where is the config file path? I want to delete all settings to test again.

**Comment by qinyuhang** [comment]
This is course by Privoxy, Simple switch to PAC or white list mode can avoid this problem. Or in Global mode update your GFW list & white list, than switch to PAC mode.
The config file is in `~/Library/Application Support/ShadowsocksX-NG/`, U can delete it all.

**Comment by iclinux** [comment]
I tried to switch to PAC or white list mode, but it didn't work at all.

Besides, It seems that ShadowsocksX-NG not only store config in "~/Library/Application\ Support/ShadowsocksX-NG/",  but also somewhere else. because after quit ShadowsocksX-NG and deleted folder "~/Library/Application\ Support/ShadowsocksX-NG/",  when restarted, ShadowsocksX-NG still remember the last config.

**Comment by qinyuhang** [comment]
U R clean a wrong PATH, SSX store config @ `~/Library/Application Support/ShadowsocksX-NG` (full PATH is `/User/yourName/Library/Application Support/ShadowsocksX-NG` or `${HOME}/Library/Application Support/ShadowsocksX-NG`) not `/Library/Application Support/ShadowsocksX-NG`

**Comment by iclinux** [comment]
NO, the path is correct, I cleaned `~/Library`, not `/Library`.

**Comment by qinyuhang** [comment]
Then there should be not config file left expect SSX helper @ `/Library/Application Support/ShadowsocksX-NG` and a launch plist @ `~/Library/LaunchAgents/com.qiuyuzhou.shadowsocksX-NG.http.plist` and `~/Library/LaunchAgents/com.qiuyuzhou.shadowsocksX-NG.local.plist`

**Comment by iclinux** [comment]
It's very strange, I deleted the files you mentioned above. and deleted Shadowsocks in application folder.

But after install Shadowsocks again, when started,  system popup a confirm dialog, said: "ShadowsocksX-NG-R8 wants to make changes. Type your password to allow this." after confirmed, Shadowsocks still cannot work well, but it did remember all the last settings such as server address, why?

**Comment by iclinux** [comment]
um, how to fix "500 Internal Privoxy Error" or "This site can’t be reached" issues? I can provide more log files if you need.

**Comment by qinyuhang** [comment]
This 500 problem is due to privoxy cannot connect. Using PAC or white list mode will avoid this problem, but you may need update your PAC file first.

**Comment by iclinux** [comment]
I've tried, but neither PC nor white list mode works. need I send the log to you?

**Comment by iclinux** [comment]
?

**Comment by qinyuhang** [comment]
The log U uploaded shows only the method and protocol U R using, not mention a error or connection, did U enabled the “Verbose Mode”?

**Comment by tuchangwei** [comment]
@iclinux  some settings are stored in "UserDefaults.standard", you can see the original settings from the source code:

`defaults.register(defaults: [
            "ShadowsocksOn": true,
            "ShadowsocksRunningMode": "auto",
            "LocalSocks5.ListenPort": NSNumber(value: 1086 as UInt16),
            "LocalSocks5.ListenAddress": "127.0.0.1",
            "PacServer.ListenAddress": "127.0.0.1",
            "PacServer.ListenPort":NSNumber(value: 8090 as UInt16),
            "LocalSocks5.Timeout": NSNumber(value: 60 as UInt),
            "LocalSocks5.EnableUDPRelay": NSNumber(value: false as Bool),
            "LocalSocks5.EnableVerboseMode": NSNumber(value: false as Bool),
            "GFWListURL": "https://raw.githubusercontent.com/gfwlist/gfwlist/master/gfwlist.txt",
            "WhiteListURL": "https://raw.githubusercontent.com/breakwa11/gfw_whitelist/master/whitelist.pac",
            "WhiteListIPURL": "https://raw.githubusercontent.com/breakwa11/gfw_whitelist/master/whiteiplist.pac",
            "AutoConfigureNetworkServices": NSNumber(value: true as Bool),
            "LocalHTTP.ListenAddress": "127.0.0.1",
            "LocalHTTP.ListenPort": NSNumber(value: 1087 as UInt16),
            "LocalHTTPOn": true,
            "LocalHTTP.FollowGlobal": true,
            "AutoCheckUpdate": false
        ])`

**Comment by R-L-T-Y** [comment]
Confirmed that cleaning  both ~/Lib/App~/Shadow* and /Lib/App~/Shadow* doesn't restore to  the default settings.
 I thought that I mess something up and hoping that I can restore everything to default.
I would be able to fetch PAC from GFWlist but fail to connect to the outside Internet via both Safari and Chrome.
That server works fluently like a butter.
Please help. If you need any more information I would be pleased to provide.
@qinyuhang

**Comment by qinyuhang** [comment]
You may try cleanup `~/.ShadowsocksX-NG/` too.

**Comment by R-L-T-Y** [comment]
No luck there. My server information stays and the situation's pretty much the same: loading progress bar freeze until "too long to respond"

**Comment by bingowrt** [comment]
Same issue i have no idea what to do

**Comment by zhaoxitao** [comment]
Hi Guys,
I found some reason of this problem, In general, if your client can not establish connect to your server, you will got this problem. And my issue is the cipher method was wrong. 

I use scan QR code setup done my server, my QR code cipher method is uppercase letter “NONE”. In the ss-local.log, I see this method dose not support. When I change it to lower case letters "none", it's work!

<img width="920" alt="2017-07-23 1 07 33" src="https://user-images.githubusercontent.com/8552883/28493198-e57f356e-6f44-11e7-8200-fae0f909d12f.png">

**Comment by qinyuhang** [comment]
@zhaoxitao fixed via https://github.com/shadowsocksr/ShadowsocksX-NG/commit/d4305d4ab465378e235cc155ab92424c0d369ab3
